src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -457,6 +460,13 @@ private unsafe bool WasEventDetected(int pollFileDescriptor, int valueFileDescri
                 int waitResult = Interop.epoll_wait(pollFileDescriptor, out epoll_event events, 1, PollingTimeout);
                 if (waitResult == -1)
                 {
+                    var errorCode = Marshal.GetLastWin32Error();
+                    if (errorCode == ERROR_CODE_EINTR)
+                    {
+                        // ignore Interrupted system call error and retry
+                        continue;
+                    }
+
                     throw new IOException("Error while waiting for pin interrupts.");
```

Comment:
Please include the error code in the message.

Reply:
@pgrawehr Do you mean the comment or exception message?

I didn't want to change the exception message in this pull request.
What about using `ExceptionHelper` for the exception (also on other places in `SysFsDriver`)?
Personally I prefer to handle that in a separate PR to keep this one simple.